app/single_resource/views.py

```diff
@@ -150,15 +151,32 @@ def edit(resource_id):
                   'form-error')
     # Field id is not needed for the form, hence omitted with [1:].
     for field_name in resource_field_names[1:]:
-        form[field_name].data = resource.__dict__[field_name]
+        if form[field_name]:
+            form[field_name].data = resource.__dict__[field_name]
     return render_template('single_resource/edit.html',
                            form=form,
                            resource_id=resource_id)
 
 
-def save_associations(resource, form, descriptors, resource_existed=True):
+def save_associations(resource, form, descriptors, resource_existed):
     """Save associations from the forms received by 'create' and 'edit' route
     handlers to the database."""
+    #first delete all the associations for this resource if it already existed (to handle the "empty" case)
+    if resource_existed:
+        options = OptionAssociation.query.filter_by(resource_id = resource.id).all()
```

Comment:
I'm deleting all the associations for the resource, regardless of if they're in the "descriptors" list passed into the function, because I wanted to handle the empty case (if you delete all the values in a field, it won't show up in the "descriptors" list). The AssociationClass can only be used for either text or option, but not for both (you have to assign it to the descriptor's type). Because I want to delete all of them, I have to do queries for both Text and Option and AssociationClass doesn't really exist/apply here.